ts/components/FiscalCodeLandscapeOverlay.tsx

```diff
@@ -52,93 +53,132 @@ const styles = StyleSheet.create({
-  }
+  const ScrollViewRef = React.createRef<ScrollView>();
 
-  public componentWillUnmount() {
-    BackHandler.removeEventListener("hardwareBackPress", this.handleBackPress);
-    // if there is an active timeout, clear it!
-    if (this.scrollTimeout !== undefined) {
-      clearTimeout(this.scrollTimeout);
-      // eslint-disable-next-line
-      this.scrollTimeout = undefined;
-    }
-  }
+  const handleBackPress = () => {
+    // On backpress the component gets unmounted, so the brightness is restored by the
+    // cleanup function
+    props.onCancel();
+    return true;
+  };
 
-  private scrollToEnd = () => {
-    if (this.props.showBackSide && this.ScrollVewRef.current) {
+  const scrollToEnd = () => {
+    if (props.showBackSide && ScrollViewRef.current) {
       // dalay the scroll to end command to wait until the ingress animation is completed
       // eslint-disable-next-line
-      this.scrollTimeout = setTimeout(() => {
-        if (this.ScrollVewRef.current) {
-          this.ScrollVewRef.current.scrollToEnd({ animated: true });
+      scrollTimeout = setTimeout(() => {
+        if (ScrollViewRef.current) {
+          ScrollViewRef.current.scrollToEnd({ animated: true });
         }
       }, 300);
     }
   };
 
-  public render() {
-    return (
-      <Container style={{ backgroundColor: customVariables.brandDarkGray }}>
-        <AppHeader noLeft={true} dark={true}>
-          <Body />
-        </AppHeader>
-        <StatusBar
-          backgroundColor={customVariables.brandDarkGray}
-          barStyle={"light-content"}
-        />
-        <ScrollView
-          style={styles.content}
-          ref={this.ScrollVewRef}
-          onLayout={this.scrollToEnd}
-        >
-          <View style={styles.headerSpacer} />
-          <View spacer={true} />
-          <View>
-            <FiscalCodeComponent
-              type={"Landscape"}
-              profile={this.props.profile}
-              getBackSide={false}
-              municipality={this.props.municipality}
-            />
-          </View>
-
-          <View spacer={true} />
+  React.useEffect(() => {
+    BackHandler.addEventListener("hardwareBackPress", handleBackPress);
+    return () => {
+      BackHandler.removeEventListener("hardwareBackPress", handleBackPress);
+      // if there is an active timeout, clear it!
+      if (scrollTimeout !== undefined) {
+        clearTimeout(scrollTimeout);
+        // eslint-disable-next-line
+        scrollTimeout = undefined;
+      }
+    };
+  }, []);
+
+  // Brightness effect manager
+  React.useEffect(() => {
+    const myBrightF = async () =>
+      await getBrightness()
+        .fold(
+          () => 0.5, // If fails, resort to a default value
+          _ => _
+        )
+        .run();
+
+    const myBright = myBrightF();
 
+    const mySetBrightF = async () =>
+      await setBrightness(HIGH_BRIGHTNESS)
+        .fold(
+          () => alert("Can't set High Brightness"),
```

Comment:
I guess alert produce no visible effects (it)
Since we don't have to show the error to the user (it can be silent handled) we could have

`const mySetBrightF = async () => await setBrightness(HIGH_BRIGHTNESS).run();`

Reply:
I agree.. Not only the  the negative case of the `TaskEither` could be useless, but I also wasn't  able to test it, as I couldn't make the `setBrightness` to failI'll fix as you suggest